webapp/controller/App.controller.js

```diff
@@ -3,36 +3,105 @@ sap.ui.define([
+	"sap/ui/core/date/UI5Date",		//
+	"../model/formatter"
+], (Device, Controller, Filter, FilterOperator, JSONModel, DateTypeRange, UI5Date, formatter) => {
 	"use strict";
 
 	return Controller.extend("sap.ui.demo.todo.controller.App", {
+		formatter: formatter,
 
 		onInit() {
+
+			let oModel = new JSONModel();
+			
+			let oOptions = {
+				"SelectedOption": "Normal(Default)",
+				"OptionCollection": [
+					{
+						"PriorityID": "5",
+						"PriName": "Top",
+						"Icon": "sap-icon://add-product"
+					},
+					{
+						"PriorityID": "4",
+						"PriName": "High",
+						"Icon": "sap-icon://add-product"
+					},
+					{
+						"PriorityID": "3",
+						"PriName": "Normal(Default)",
+						"Icon": "sap-icon://add-product"
+					},
+					{
+						"PriorityID": "2",
+						"PriName": "Low",
+						"Icon": "sap-icon://add-product"
+					},
+					{
+						"PriorityID": "1",
+						"PriName": "Very Low",
+						"Icon": "sap-icon://add-product"
+					}
+				]
+			};
+
 			this.aSearchFilters = [];
 			this.aTabFilters = [];
+			this.aHashTagFilters = [];
 
-			this.getView().setModel(new JSONModel({
+			oModel.setData({
 				isMobile: Device.browser.mobile,
+				selectOptions: oOptions,
+				//valueDP1: UI5Date.getInstance(),
 				filterText: undefined
-			}), "view");
+			});
+
+			this.getView().setModel(oModel, "view");
+			// should do it on component level.
+			this.getView().getModel().setProperty("/default_newTodo/DDLAtUTC", JSON.stringify(UI5Date.getInstance()).slice(1,-1))
+            this.getView().getModel().setProperty("/newTodo/DDLAtUTC", JSON.stringify(UI5Date.getInstance()).slice(1,-1))
 		},
 
 		/**
 		 * Adds a new todo item to the bottom of the list.
 		 */
 		addTodo() {
 			const oModel = this.getView().getModel();
+			if(!oModel.getProperty("/newTodo/title"))
+			{
+				return
+			}
+			
 			const aTodos = oModel.getProperty("/todos").map((oTodo) => Object.assign({}, oTodo));
 
+			let DateofAddedUTC = JSON.stringify(UI5Date.getInstance()).slice(1,-1)
+
+			// Hansen:
+			// todo: fix the problem of Date(UTC...)
+			const newTodo = oModel.getProperty("/newTodo")
+			const defaultNewTodo = oModel.getProperty("/default_newTodo")
+			let newTodoHashTagIdx = String(newTodo.title).search("#")
+			if(newTodoHashTagIdx === -1) {
+				newTodoHashTagIdx = String(newTodo.title).length
+			}
+			let newTodoTitle = newTodo.title.substring(0, newTodoHashTagIdx)
+			let newTodoHashTag = newTodo.title.substring(newTodoHashTagIdx)
+			let newTodoDDL = newTodo.DDLAtUTC
 			aTodos.push({
-				title: oModel.getProperty("/newTodo"),
-				completed: false
+				title: newTodoTitle,
+				hashTag: newTodoHashTag,
+				completed: false,
+				priority: newTodo.priority,  //newTodoPriorityIndex[newTodo.priority],
+				DDLAtUTC: newTodoDDL,
+				addedAtUTC: DateofAddedUTC,
+				//addedAt: Date(DateofAddedUTC)
 			});
 
 			oModel.setProperty("/todos", aTodos);
-			oModel.setProperty("/newTodo", "");
+			oModel.setProperty("/newTodo", defaultNewTodo);
```

Comment:
maybe a nice comment that you reset the model with a new semi-empty todo would be nice here :)